{
    "GSD": {
        "alias": "CVE-2019-20477",
        "description": "PyYAML 5.1 through 5.1.2 has insufficient restrictions on the load and load_all functions because of a class deserialization issue, e.g., Popen is a class in the subprocess module. NOTE: this issue exists because of an incomplete fix for CVE-2017-18342.",
        "id": "GSD-2019-20477",
        "references": [
            "https://www.suse.com/security/cve/CVE-2019-20477.html",
            "https://access.redhat.com/errata/RHSA-2021:0420",
            "https://access.redhat.com/errata/RHSA-2020:4641",
            "https://linux.oracle.com/cve/CVE-2019-20477.html"
        ]
    },
    "namespaces": {
        "cve.org": {
            "CVE_data_meta": {
                "ASSIGNER": "cve@mitre.org",
                "ID": "CVE-2019-20477",
                "STATE": "PUBLIC"
            },
            "affects": {
                "vendor": {
                    "vendor_data": [
                        {
                            "product": {
                                "product_data": [
                                    {
                                        "product_name": "n/a",
                                        "version": {
                                            "version_data": [
                                                {
                                                    "version_value": "n/a"
                                                }
                                            ]
                                        }
                                    }
                                ]
                            },
                            "vendor_name": "n/a"
                        }
                    ]
                }
            },
            "data_format": "MITRE",
            "data_type": "CVE",
            "data_version": "4.0",
            "description": {
                "description_data": [
                    {
                        "lang": "eng",
                        "value": "PyYAML 5.1 through 5.1.2 has insufficient restrictions on the load and load_all functions because of a class deserialization issue, e.g., Popen is a class in the subprocess module. NOTE: this issue exists because of an incomplete fix for CVE-2017-18342."
                    }
                ]
            },
            "problemtype": {
                "problemtype_data": [
                    {
                        "description": [
                            {
                                "lang": "eng",
                                "value": "n/a"
                            }
                        ]
                    }
                ]
            },
            "references": {
                "reference_data": [
                    {
                        "name": "https://github.com/yaml/pyyaml/blob/master/CHANGES",
                        "refsource": "MISC",
                        "url": "https://github.com/yaml/pyyaml/blob/master/CHANGES"
                    },
                    {
                        "name": "https://www.exploit-db.com/download/47655",
                        "refsource": "MISC",
                        "url": "https://www.exploit-db.com/download/47655"
                    },
                    {
                        "name": "FEDORA-2020-3162499159",
                        "refsource": "FEDORA",
                        "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/52N5XS73Z5S4ZN7I7R56ICCPCTKCUV4H/"
                    },
                    {
                        "name": "FEDORA-2020-e84e90dc4a",
                        "refsource": "FEDORA",
                        "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/33VBUY73AA6CTTYL3LRWHNFDULV7PFPN/"
                    }
                ]
            }
        },
        "gitlab.com": {
            "advisories": [
                {
                    "affected_range": ">=5.1,<5.2",
                    "affected_versions": "All versions starting from 5.1 before 5.2",
                    "cvss_v2": "AV:N/AC:L/Au:N/C:P/I:P/A:P",
                    "cvss_v3": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H",
                    "cwe_ids": [
                        "CWE-1035",
                        "CWE-502",
                        "CWE-937"
                    ],
                    "date": "2022-01-04",
                    "description": "PyYAML 5.1 through 5.1.2 has insufficient restrictions on the load and load_all functions because of a class deserialization issue, e.g., Popen is a class in the subprocess module. NOTE: this issue exists because of an incomplete fix for CVE-2017-18342.",
                    "fixed_versions": [
                        "5.2"
                    ],
                    "identifier": "CVE-2019-20477",
                    "identifiers": [
                        "GHSA-3pqx-4fqf-j49f",
                        "CVE-2019-20477"
                    ],
                    "not_impacted": "All versions before 5.1, all versions starting from 5.2",
                    "package_slug": "pypi/PyYAML",
                    "pubdate": "2021-04-20",
                    "solution": "Upgrade to version 5.2 or above.",
                    "title": "Deserialization of Untrusted Data",
                    "urls": [
                        "https://nvd.nist.gov/vuln/detail/CVE-2019-20477",
                        "https://github.com/yaml/pyyaml/blob/master/CHANGES",
                        "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/33VBUY73AA6CTTYL3LRWHNFDULV7PFPN/",
                        "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/52N5XS73Z5S4ZN7I7R56ICCPCTKCUV4H/",
                        "https://www.exploit-db.com/download/47655",
                        "https://github.com/advisories/GHSA-3pqx-4fqf-j49f"
                    ],
                    "uuid": "43a0466c-7d41-47b6-aafa-f6c2cfaaef4d"
                }
            ]
        },
        "nvd.nist.gov": {
            "configurations": {
                "CVE_data_version": "4.0",
                "nodes": [
                    {
                        "children": [],
                        "cpe_match": [
                            {
                                "cpe23Uri": "cpe:2.3:a:pyyaml:pyyaml:*:*:*:*:*:*:*:*",
                                "cpe_name": [],
                                "versionEndIncluding": "5.1.2",
                                "versionStartIncluding": "5.1",
                                "vulnerable": true
                            }
                        ],
                        "operator": "OR"
                    },
                    {
                        "children": [],
                        "cpe_match": [
                            {
                                "cpe23Uri": "cpe:2.3:o:fedoraproject:fedora:30:*:*:*:*:*:*:*",
                                "cpe_name": [],
                                "vulnerable": true
                            },
                            {
                                "cpe23Uri": "cpe:2.3:o:fedoraproject:fedora:31:*:*:*:*:*:*:*",
                                "cpe_name": [],
                                "vulnerable": true
                            }
                        ],
                        "operator": "OR"
                    }
                ]
            },
            "cve": {
                "CVE_data_meta": {
                    "ASSIGNER": "cve@mitre.org",
                    "ID": "CVE-2019-20477"
                },
                "data_format": "MITRE",
                "data_type": "CVE",
                "data_version": "4.0",
                "description": {
                    "description_data": [
                        {
                            "lang": "en",
                            "value": "PyYAML 5.1 through 5.1.2 has insufficient restrictions on the load and load_all functions because of a class deserialization issue, e.g., Popen is a class in the subprocess module. NOTE: this issue exists because of an incomplete fix for CVE-2017-18342."
                        }
                    ]
                },
                "problemtype": {
                    "problemtype_data": [
                        {
                            "description": [
                                {
                                    "lang": "en",
                                    "value": "CWE-502"
                                }
                            ]
                        }
                    ]
                },
                "references": {
                    "reference_data": [
                        {
                            "name": "https://www.exploit-db.com/download/47655",
                            "refsource": "MISC",
                            "tags": [
                                "Exploit",
                                "Third Party Advisory",
                                "VDB Entry"
                            ],
                            "url": "https://www.exploit-db.com/download/47655"
                        },
                        {
                            "name": "https://github.com/yaml/pyyaml/blob/master/CHANGES",
                            "refsource": "MISC",
                            "tags": [
                                "Release Notes",
                                "Third Party Advisory"
                            ],
                            "url": "https://github.com/yaml/pyyaml/blob/master/CHANGES"
                        },
                        {
                            "name": "FEDORA-2020-3162499159",
                            "refsource": "FEDORA",
                            "tags": [
                                "Mailing List",
                                "Third Party Advisory"
                            ],
                            "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/52N5XS73Z5S4ZN7I7R56ICCPCTKCUV4H/"
                        },
                        {
                            "name": "FEDORA-2020-e84e90dc4a",
                            "refsource": "FEDORA",
                            "tags": [
                                "Mailing List",
                                "Third Party Advisory"
                            ],
                            "url": "https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/33VBUY73AA6CTTYL3LRWHNFDULV7PFPN/"
                        }
                    ]
                }
            },
            "impact": {
                "baseMetricV2": {
                    "acInsufInfo": false,
                    "cvssV2": {
                        "accessComplexity": "LOW",
                        "accessVector": "NETWORK",
                        "authentication": "NONE",
                        "availabilityImpact": "PARTIAL",
                        "baseScore": 7.5,
                        "confidentialityImpact": "PARTIAL",
                        "integrityImpact": "PARTIAL",
                        "vectorString": "AV:N/AC:L/Au:N/C:P/I:P/A:P",
                        "version": "2.0"
                    },
                    "exploitabilityScore": 10.0,
                    "impactScore": 6.4,
                    "obtainAllPrivilege": false,
                    "obtainOtherPrivilege": false,
                    "obtainUserPrivilege": false,
                    "severity": "HIGH",
                    "userInteractionRequired": false
                },
                "baseMetricV3": {
                    "cvssV3": {
                        "attackComplexity": "LOW",
                        "attackVector": "NETWORK",
                        "availabilityImpact": "HIGH",
                        "baseScore": 9.8,
                        "baseSeverity": "CRITICAL",
                        "confidentialityImpact": "HIGH",
                        "integrityImpact": "HIGH",
                        "privilegesRequired": "NONE",
                        "scope": "UNCHANGED",
                        "userInteraction": "NONE",
                        "vectorString": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H",
                        "version": "3.1"
                    },
                    "exploitabilityScore": 3.9,
                    "impactScore": 5.9
                }
            },
            "lastModifiedDate": "2022-01-01T19:38Z",
            "publishedDate": "2020-02-19T04:15Z"
        }
    }
}